src/backend/distributed/commands/role.c

```diff
@@ -878,7 +878,9 @@ GenerateGrantRoleStmtsOfRole(Oid roleid)
 		if (membership->admin_option)
 		{
 			DefElem *opt = makeDefElem("admin", (Node *) makeBoolean(true), -1);
-			grantRoleStmt->opt = list_make1(opt);
+			DefElem *inherit_opt = makeDefElem("inherit", (Node *) makeBoolean(true), -1);
```

Comment:
As of PG16, pg_auth_members entries now have `inherit_option` and `set_option` as well. So, here we should look at those separately, and not add the inherit and set options under admin_option.